initdb.d/04_grants.sql

```diff
@@ -1,29 +1,162 @@
+              ) ON sda.files TO ingest;
+GRANT INSERT ON sda.checksums TO ingest;
+GRANT UPDATE ON sda.checksums TO ingest;
+GRANT SELECT ON sda.checksums TO ingest;
+GRANT USAGE, SELECT ON SEQUENCE sda.checksums_id_seq TO ingest;
+
+--------------------------------------------------------------------------------
+
+CREATE ROLE verify;
+-- uses: db.GetHeader, and db.MarkCompleted
+GRANT USAGE ON SCHEMA sda TO verify;
+GRANT SELECT ON sda.files TO verify;
+GRANT UPDATE (archive_file_path, archive_file_size, decrypted_file_size) ON sda.files TO verify;
+GRANT INSERT ON sda.checksums TO verify;
+GRANT USAGE, SELECT ON SEQUENCE sda.checksums_id_seq TO verify;
+GRANT INSERT ON sda.file_event_log TO verify;
+GRANT USAGE, SELECT ON SEQUENCE sda.file_event_log_id_seq TO verify;
+
+-- legacy schema
+GRANT USAGE ON SCHEMA local_ega TO verify;
+GRANT SELECT ON local_ega.files TO verify;
+GRANT SELECT ON local_ega.main_to_files TO verify;
+GRANT SELECT ON local_ega.status_translation TO verify;
+GRANT UPDATE (archive_file_reference, decrypted_file_size, status) ON local_ega.main TO verify;
+GRANT UPDATE (status, archive_filesize, archive_file_checksum,
+              archive_file_checksum_type, decrypted_file_size,
+              decrypted_file_checksum, decrypted_file_checksum_type) ON local_ega.files TO verify;
+GRANT UPDATE (stable_id, submission_user, submission_file_path, submission_file_size,
+              archive_file_path, archive_file_size, decrypted_file_size, header, encryption_method
+              ) ON sda.files TO verify;
+GRANT INSERT ON sda.checksums TO verify;
+GRANT UPDATE ON sda.checksums TO verify;
+GRANT SELECT ON sda.checksums TO verify;
+GRANT USAGE, SELECT ON SEQUENCE sda.checksums_id_seq TO verify;
+
+--------------------------------------------------------------------------------
+
+CREATE ROLE finalize;
+-- uses: db.MarkReady
+GRANT USAGE ON SCHEMA sda TO finalize;
+GRANT UPDATE (stable_id) ON sda.files TO finalize;
+GRANT SELECT ON sda.files TO finalize;
+GRANT SELECT ON sda.checksums TO finalize;
+GRANT INSERT ON sda.file_event_log TO finalize;
+GRANT USAGE, SELECT ON SEQUENCE sda.file_event_log_id_seq TO finalize;
+
+-- legacy schema
+GRANT USAGE ON SCHEMA local_ega TO finalize;
+GRANT SELECT ON local_ega.main_to_files TO finalize;
+GRANT SELECT ON local_ega.status_translation TO finalize;
+GRANT UPDATE (stable_id, status) ON local_ega.files TO finalize;
+GRANT SELECT ON local_ega.files TO finalize;
+GRANT UPDATE (status, archive_filesize, archive_file_checksum,
+              archive_file_checksum_type, decrypted_file_size,
+              decrypted_file_checksum, decrypted_file_checksum_type) ON local_ega.files TO finalize;
+GRANT UPDATE (stable_id, submission_user, submission_file_path, submission_file_size,
+              archive_file_path, archive_file_size, decrypted_file_size, header, encryption_method
+              ) ON sda.files TO finalize;
```

Comment:
`sda.files` shouldn't be part of the `legacy schema`

Reply:
These permissions are needed for the legacy views, so that's why they're grouped with the legacy layer! :)